api/tests/src/legacy_soap_api/applicants/services/test_get_opportunity_list_response.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+
+from src.db.models.competition_models import Competition
+from src.db.models.opportunity_models import Opportunity, OpportunityAssistanceListing
+from src.legacy_soap_api.applicants.schemas import (
+    CFDADetails,
+    GetOpportunityListRequest,
+    OpportunityFilter,
+)
+from src.legacy_soap_api.applicants.services.get_opportunity_list_response import (
+    get_opportunity_list_response,
+)
+from tests.lib.db_testing import cascade_delete_from_db_table
+from tests.src.db.models.factories import (
+    CompetitionFactory,
+    OpportunityAssistanceListingFactory,
+    OpportunityFactory,
+)
+
+
+@pytest.fixture(autouse=True)
+def truncate_competitions(db_session):
+    # This will truncate the competitions and related data for each test within this test file.
+    cascade_delete_from_db_table(db_session, Competition)
+    cascade_delete_from_db_table(db_session, Opportunity)
+    cascade_delete_from_db_table(db_session, OpportunityAssistanceListing)
```

Comment:
I think I might know how to remove at least one of these cascade deletes for the assistance listing table

In our CompetitionFactory change the line where we add an opportunity_assistance_listing to:
```py
    opportunity_assistance_listing = factory.SubFactory(OpportunityAssistanceListingFactory, opportunity=factory.SelfAttribute("..opportunity"))
```
Right now the competition factory makes an opportunity assistance listing with a random different opportunity than the one associated with the competition, so some stuff gets orphaned. This fixes that.